simplipy/api.py

```diff
@@ -270,14 +295,17 @@ def is_fatal_error(err: ClientResponseError) -> bool:
                 which is where this error can occur; we can't control when/how that
                 happens (e.g., we might query the API in the middle of a base station
                 update), so it should be viewed as retryable.
+        3. 404: When fetching media files, you may get a 404 if the media file is not
+                yet available to read.  Keep trying however, and it will eventually
+                return a 200.
 
         Args:
             err: An ``aiohttp`` ``ClientResponseError``
 
         Returns:
             Whether the error is a fatal one.
         """
-        if err.status in (401, 409):
+        if err.status in (401, 409, 404):
```

Comment:
This hasn't been resolved. We shouldn't catch all 404s like this.

Reply:
My apologies,  over the holidays I forgot I was in the middle of resolving some lint errors.  I just committed api.py that should fix this.  There are two "apis" now; the original one that calls the original SS apis, and a new one that fetches media with absolute S3 urls.  The latter still must retry on 404s, and does so for 10 attempts before giving up.  The original api function will fail on 404 like it used to.

I still try to share as much code as possible for auth token handling, refresh and backoffs.